apps/wpcom-block-editor/src/common/tracking.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * External dependencies
+ */
+import { use, select } from '@wordpress/data';
+import { registerPlugin } from '@wordpress/plugins';
+import { castArray } from 'lodash';
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import tracksRecordEvent from './tracking/track-record-event';
+import delegateEventTracking from './tracking/delegate-event-tracking';
+
+// Debugger.
+const debug = debugFactory( 'wpcom-block-editor:tracking' );
+
+// In case Tracks hasn't loaded.
+if ( typeof window !== 'undefined' ) {
+	window._tkq = window._tkq || [];
+}
+
+/**
+ * Looks up the block name based on its id.
+ *
+ * @param {string} blockId Blog identifier.
+ * @return {string|null} Blg name if it exists. Otherwise, `null`.
+ */
+const getTypeForBlockId = blockId => {
+	const block = select( 'core/block-editor' ).getBlock( blockId );
+	return block ? block.name : null;
+};
+
+/**
+ * A lot of actions accept either string (block id)
+ * or an array of multiple string when operating with multiple blocks selected.
+ * This is a convenience method that processes the first argument of the action (blocks)
+ * and calls your tracking for each of the blocks involved in the action.
+ *
+ * @param {string} eventName event name
+ * @return {function} track handler
+ */
+const getBlocksTracker = eventName => blockIds => {
+	// track separately for each block
+	castArray( blockIds ).forEach( blockId => {
+		tracksRecordEvent( eventName, { block_name: getTypeForBlockId( blockId ) } );
+	} );
+};
+
+/**
+ * Track block insertion.
+ *
+ * @param {object|array} blocks block instance object or an array of such objects
+ * @return {void}
+ */
+const trackBlockInsertion = blocks => {
+	castArray( blocks ).forEach( block => {
+		tracksRecordEvent( 'wpcom_block_picker_block_inserted', {
```

Comment:
This guide is a good read whenever adding new events: PCYsg-4S2-p2

In particular, I think it's important to try to adhere to the naming convention:

```
<source>_<context>_<optional subcontext>_<action>_<optional qualifier>
```

`wpcom` may be OK for a source, but we use that for events from the WPCOM backend which may be confusing. @ockham raises an important question, let's [discuss on that thread](https://github.com/Automattic/wp-calypso/pull/34655/files#r304745916).

I think the _context_ here should be `blockpicker`.

`block` seems like a fine _subcontext_ and `inserted` like a good _action_.

I'd suggest:

```
(to be determined)_blockpicker_block_inserted
```

Reply:
This event name was changed as it no longer involves block picker. It will be just `wpcom_block_inserted` and is tracking all block insertions, not just through the picker. I think that will be good for your funnel as it will cover all possible ways a person can access the block.

I also want to point out we were aware of the field guide and `wpcom_block_picker_block_inserted` fully follows the naming schema. Read the page fully and you'll see multiple words with underscores are allowed in all parts and are recommended in examples.